### PR TITLE
[CDAP-17248] Added File Encoding support for GCS Source Plugin

### PR DESCRIPTION
**Pull request by fernst**
This PR Depends on cdapio/hydrator-plugins#1255

### CODE REVIEW

File: src/main/java/io/cdap/plugin/gcp/gcs/source/GCSSource.java [hunk 1]
```diff
@@ -320,6 +337,11 @@ public boolean skipHeader() {
       return skipHeader == null ? false : skipHeader;
     }
 
+    @Nullable
+    public String getFileEncoding() {
```
Comment (yaojiefeng) [inline src/main/java/io/cdap/plugin/gcp/gcs/source/GCSSource.java:341]: Just curious here, from https://github.com/cdapio/hydrator-plugins/pull/1255/files, I saw this method is only used in the validation, how does the input format knows which encoding is or am I missing anything

**Review by yaojiefeng — COMMENTED** [review]
One comment

File: src/main/java/io/cdap/plugin/gcp/gcs/source/GCSSource.java (hunk 1 above)
Comment (fernst) [reply src/main/java/io/cdap/plugin/gcp/gcs/source/GCSSource.java:341]: @yaojiefeng The final piece is this PR in `core-plugins`. It's not part of the original PR as these are 2 modules with cross dependencies and I can't really build both of them on the same PR https://github.com/cdapio/hydrator-plugins/pull/1297

**Review by yaojiefeng — APPROVED** [review]

**Pull request merged**